modules/account-lib/src/coin/dot/addProxyBuilder.ts

```diff
@@ -0,0 +1,142 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { decode, methods } from '@substrate/txwrapper-polkadot';
+import BigNumber from 'bignumber.js';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TransactionType } from '../baseCoin';
+import { AddProxyArgs, MethodNames, proxyType } from './iface';
+import { AddProxyTransactionSchema } from './txnSchema';
+
+export class AddProxyBuilder extends TransactionBuilder {
+  protected _delegate: string;
+  protected _proxyType: proxyType;
+  protected _delay: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /**
+   *
+   * Register a proxy account for the sender that is able to make calls on its behalf.
+   *
+   * @returns {UnsignedTransaction} an unsigned Dot transaction
+   *
+   * @see https://polkadot.js.org/docs/substrate/extrinsics/#proxy
+   */
+  protected buildDotTxn(): UnsignedTransaction {
+    const baseTxInfo = this.createBaseTxInfo();
+    return methods.proxy.addProxy(
+      {
+        delegate: this._delegate,
+        proxyType: this._proxyType,
+        delay: this._delay,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.AddProxy;
+  }
+
+  /**
+   *
+   * The account to delegate auth to
+   *
+   * @param {string} delegate
+   * @returns {AddProxyBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#why-use-a-proxy
+   */
+  delegate(delegate: string): this {
```

Comment:
`delegate` is too coin specific and it is not clear (it can be misunderstood with staking). Rename to `owner`